app/src/components/adminConsole/AdminConsole.tsx

```diff
@@ -1,38 +1,32 @@
 import React, { useState, useEffect, useCallback } from "react";
 import SnappableManager from "./SnappableManager";
 import CurrentCup from "./currentCup/CurrentCup";
-import { getExistsUnpublished } from "../../firebase/cups/CupService";
+import { getCurrentCupIfExists } from "../../firebase/cups/CupService";
+import Cup from "../../types/Cup";
 
 const AdminConsole = () => {
-    const [isCup, setIsCup] = useState<Boolean>(false);
+    const [cup, setCup] = useState<Cup | undefined>(false);
```

Comment:
I would like to avoid the necessity for ` | undefined`, it's a bit unpleasant.

I would like `AdminConsole` to get a list of unpublished cups from Firestore.
(We know this list will contain at most a single cup.)

If the list is empty we render the 'create new cup' component.
If the list is _not_ empty, we render the first element.

This will require some more extensive refactoring but this would a) be a good exercise, and b) be useful for future development.

Reply:
Has now been done :)